integration/src/main/java/scripts/testscripts/SnapshotDatasetOperations.java

```diff
@@ -0,0 +1,165 @@
+            .addContentsItem(
+                new SnapshotRequestContentsModel()
+                    .datasetName(TEST_DATASET_NAME)
+                    .mode(SnapshotRequestContentsModel.ModeEnum.BYFULLVIEW));
+    var createJobId = snapshotsApi.createSnapshot(request).getId();
+
+    var jobApi = new JobsApi(datarepoClient);
+    JobModel job;
+    do {
+      job = jobApi.retrieveJob(createJobId);
+    } while (job.getJobStatus() == JobModel.JobStatusEnum.RUNNING);
+
+    if (job.getJobStatus() != JobModel.JobStatusEnum.SUCCEEDED) {
+      throw new RuntimeException("Create snapshot failed: " + job);
+    }
+    @SuppressWarnings("unchecked")
+    Map<Object, Object> result = (Map<Object, Object>) jobApi.retrieveJobResult(createJobId);
+    snapshotId = UUID.fromString((String) result.get("id"));
+    log.info("created snapshot " + snapshotId);
+  }
+
+  private static final String METADATA_1 = """
+      {"name": "test"}""";
+
+  private static final String METADATA_2 = """
+      {"name": "test2"}""";
+
+  @Override
+  public void userJourney(TestUserSpecification testUser) throws Exception {
+    var client = new CatalogClient(server, testUser);
+    datasetsApi = new DatasetsApi(client);
+
+    // Given a snapshot, create a catalog entry.
+    var request =
+        new CreateDatasetRequest()
+            .catalogEntry(METADATA_1)
+            .storageSourceId(snapshotId.toString())
+            .storageSystem(StorageSystem.TDR);
+    datasetId = datasetsApi.createDataset(request).getId();
+    assertThat(client.getStatusCode(), is(HttpStatusCodes.STATUS_CODE_OK));
+    assertThat(datasetId, notNullValue());
+    log.info("created dataset " + datasetId);
+
+    // Retrieve the entry
+    assertThat(datasetsApi.getDataset(datasetId), is(METADATA_1));
+    assertThat(client.getStatusCode(), is(HttpStatusCodes.STATUS_CODE_OK));
+
+    // Retrieve all datasets
+    var datasets = datasetsApi.listDatasets();
+    assertThat(client.getStatusCode(), is(HttpStatusCodes.STATUS_CODE_OK));
+    validateListResult(datasets.getResult());
+
+    // Modify the entry
+    datasetsApi.updateDataset(METADATA_2, datasetId);
+    assertThat(client.getStatusCode(), is(HttpStatusCodes.STATUS_CODE_NO_CONTENT));
+
+    // Verify modify success
+    assertThat(datasetsApi.getDataset(datasetId), is(METADATA_2));
+    assertThat(client.getStatusCode(), is(HttpStatusCodes.STATUS_CODE_OK));
+
+    // Delete the entry
+    datasetsApi.deleteDataset(datasetId);
+    assertThat(client.getStatusCode(), is(HttpStatusCodes.STATUS_CODE_NO_CONTENT));
+
+    // Verify delete success.
+    var apiException = assertThrows(ApiException.class, () -> datasetsApi.getDataset(datasetId));
+    assertThat(apiException.getCode(), is(HttpStatusCodes.STATUS_CODE_NOT_FOUND));
+    assertThat(apiException.getResponseBody(), containsString(datasetId.toString()));
+    datasetId = null;
+  }
+
+  private void validateListResult(List<Object> listResult) {
```

Comment:
```suggestion
  private void datasetsHaveEntriesAndRoles(List<Object> datasetList) {
```

Reply:
This is asserting that the result of the list operation contains a specific dataset with specific attributes. So maybe `resultHasDatasetWithRoles()`? I guess in general I think about method names as being <action><thing> so "validate" is the action and "list result" is the thing. Also I think about a `has` method as usually returning a `boolean`, which this doesn't do.

Reply:
That sounds reasonable 👍